File: src/main/java/work/lclpnet/ap2/impl/game/item/SpecialItemContext.java
package work.lclpnet.ap2.impl.game.item;

import net.minecraft.item.ItemStack;
import net.minecraft.server.network.ServerPlayerEntity;
import org.jetbrains.annotations.Nullable;
import work.lclpnet.kibu.scheduler.api.TaskScheduler;

public interface SpecialItemContext {

    void removeSpecialItem(ServerPlayerEntity player, SpecialItem item);

    boolean isSpecialItem(ItemStack stack, @Nullable SpecialItem item);

    boolean hasSpecialItem(ServerPlayerEntity player, @Nullable SpecialItem item);

    TaskScheduler scheduler();
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/item/SpecialItemContext.kt
package work.lclpnet.ap2.game.item

import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.item.ItemStack
import work.lclpnet.kibu.scheduler.api.TaskScheduler
import work.lclpnet.kibu.translate.Translations

interface SpecialItemContext {

    val scheduler: TaskScheduler

    val translations: Translations

    fun removeSpecialItem(player: ServerPlayer, item: SpecialItem)

    fun isSpecialItem(stack: ItemStack, item: SpecialItem): Boolean

    fun hasSpecialItem(player: ServerPlayer, item: SpecialItem?): Boolean
}
